### ISSUE TITLE
Attempting to access /admin fails due to too many redirects

### ISSUE THREAD

**Issue by puppystack** [bitnami/bitnami-docker-magento#86]
Using https://raw.githubusercontent.com/bitnami/bitnami-docker-magento/master/docker-compose.yml and bringing the system up - the homepage is fine, but attempting to access http://127.0.0.1/admin results in:

```
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:13 +0000] "GET /admin HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:13 +0000] "GET /index.php/admin/admin/index/index/key/b6105c4c7e5571e0e12630c8569cc45352c1ea1614c4e0804d65c02b006767f3/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:14 +0000] "GET /index.php/admin/admin/index/index/key/2fddae46a34dcb40dafe95e4b71bea7dd02bfa6d38eb5740e0fe9f204ebae865/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:14 +0000] "GET /index.php/admin/admin/index/index/key/aa4b33fc97aac808879276eb08c5328f25e79883055bb6e5520c6dc9a14fb72a/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:15 +0000] "GET /index.php/admin/admin/index/index/key/0583655040c6eba6ae9fe5dfce42f5b6f785177df25647a341f170ebdf038efb/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:15 +0000] "GET /index.php/admin/admin/index/index/key/a5e04a909d14521b3d850803712ad9494e363949e4c283016ccc05f1fb692eb3/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:16 +0000] "GET /index.php/admin/admin/index/index/key/187381e0f13d035bb4a44c704751c37b704dfda21ef66b97d9727ff2174f66b9/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:16 +0000] "GET /index.php/admin/admin/index/index/key/d8334b57d6b30f4f7022d7e29eb0952f7a7284d222c7dcd2c3c7c5c8bc44f46f/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:17 +0000] "GET /index.php/admin/admin/index/index/key/3b148db99a47e317af8667ecc88ea8aeb2e3590c1ad8317c012e7f0e190a62ed/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:17 +0000] "GET /index.php/admin/admin/index/index/key/d16af5844ab2cc6c9ec6f14e08e762f149e4aefb1a3d335873f15e446500d74c/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:18 +0000] "GET /index.php/admin/admin/index/index/key/ad8e9697d1f243c2edf6c14bf36ee984e66464b2c32a8668e9422a2a4ead5081/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:18 +0000] "GET /index.php/admin/admin/index/index/key/c81b37879cbfcded296f8fda4356fe6a4cd8e849e3d322d67962b09779ea1562/ HTTP/1.1" 302 -
magento_1  | 172.19.0.1 - - [17/Jul/2018:08:41:19 +0000] "GET /index.php/admin/admin/index/index/key/91fcdb6db289fec2745d495ad90a5de4c05ce4211be58fc5181154d5cb6b96e6/ HTTP/1.1" 302 -
[... ~30 similar entries in total ...]
```
which causes browsers to issue a too many redirects error and give up.

**Comment by tompizmor** [comment]
Hello @puppystack,

Can you give us more details about your docker environment? I just tried and it worked fine for me.

Are you using docker in a linux machine or are you using docker for mac or for windows?

Regards,
Tomas

**Comment by zhukovra** [comment]
@tompizmor hi! I don't know why had redirects yesterday, but now I've recreated containers and all works good. Local docker without any additional env. variables.

**Comment by marcosbc** [comment]
Thanks for the update @zhukovra! Let us know if you find a similar issue in the future.

**Comment by j3k0** [comment]
Similar issue here. I was using SSL, solved with this answer: https://magento.stackexchange.com/a/211214

**Comment by j3k0** [comment]
I wasn't that, I did configure `MAGENTO_HOST`.

The problem as I understand it was that the configuration variable `web/secure/use_in_adminhtml` was set to `0`. However, my magento deployment lies behind a proxy that redirects http to https.

Cleaning the cache after changing the config was also necessary.

Maybe that can can be solved automatically with a env-var like `MAGENTA_FORCE_ADMIN_SSL`.

**Comment by tompizmor** [comment]
Hello @j3k0,

Thanks for sharing your findings. We are aware of this problem as it has come up in these other issues https://github.com/bitnami/bitnami-docker-magento/issues/64 https://github.com/bitnami/bitnami-docker-magento/issues/68.

Adding the `MAGENTO_FORCE_ADMIN_SSL` env-var is a valid option so we may add that configuration option soon.

**Comment by rkmax** [comment]
I have the same issue, setting the variable `MAGENTO_HOST=localhost` make the thing work but still is redirecting wrongly
![imagen](https://user-images.githubusercontent.com/395707/48913868-b6194c00-ee47-11e8-92a1-40e4b918fb6f.png)

**Comment by tompizmor** [comment]
Hi @rkmax,

Did you check the issue https://github.com/bitnami/bitnami-docker-magento/issues/64?

More precisely these two comments:
1. https://github.com/bitnami/bitnami-docker-magento/issues/64#issuecomment-343065461
2. https://github.com/bitnami/bitnami-docker-magento/issues/64#issuecomment-364164266

**Comment by rkmax** [comment]
@tompizmor, thank you I will check that

**Comment by Braintelligence** [comment]
You mean the shell command via Magento cli?

**Comment by tompizmor** [comment]
Yes, the SQL and PHP commands. Some users have provided the following workarounds.

1. https://github.com/bitnami/bitnami-docker-magento/issues/64#issuecomment-343065461 and https://github.com/bitnami/bitnami-docker-magento/issues/64#issuecomment-364164266

2. https://github.com/bitnami/bitnami-docker-magento/issues/64#issuecomment-442812918

3. https://github.com/bitnami/bitnami-docker-magento/issues/64#issuecomment-449640389

**Comment by marcosbc** [comment]
Hi all, we have just released a new major release of the Bitnami Docker image for Magento, [2.4.1-debian-10-r80](https://github.com/bitnami/bitnami-docker-magento#241-debian-10-r80). It includes major changes with respect to how the image is initialized, and now the source code for the initialization scripts is also published in the [rootfs](https://github.com/bitnami/bitnami-docker-magento/tree/master/2/debian-10/rootfs) folder of this repo.

As part of these changes, we added support for the `MAGENTO_ENABLE_HTTPS` and `MAGENTO_ENABLE_ADMIN_HTTPS` environment which let you decide where to enable HTTPS. By default it will be disabled in both store and admin respectively.
